monitorcontroller/monitorcontroller.go

```diff
@@ -135,27 +208,37 @@ func (monitor *MonitorController) CloudDisconnected() {
  **********************************************************************************************************************/
 
 func (monitor *MonitorController) processQueue(ctx context.Context) {
-	sendTicker := time.NewTicker(MinSendPeriod)
+	sendTicker := time.NewTicker(monitor.maxSendInterval.Duration)
 
 	for {
 		select {
 		case <-sendTicker.C:
 			monitor.Lock()
+			monitor.sendMessages()
+			monitor.Unlock()
```

Comment:
Better to move lock/unlock inside `sendMessages`

Reply:
done